server/views/pages/goal/create/index.njk

```diff
@@ -23,37 +23,84 @@
-          <input type="hidden" name="_csrf" value="{{ csrfToken }}" />
-          <input type="hidden" name="prisonNumber" value="{{ form.prisonNumber }}" />
-
-          {% set titleLabelHtml %}
-            Describe the goal you want to create for {{ prisonerSummary.firstName }} {{ prisonerSummary.lastName }}
-          {% endset %}
-
-          {% set hintTextHtml %}
-            <p class="govuk-hint">Goals should be SMART, meaning 'specific', 'measurable', 'achievable', 'relevant' and 'time-bound'. SMART goals allow progress to be better measured and evaluated.</p>
-            <p class="govuk-hint">You can add the steps to achieve the goal on the next page.</p>
-          {% endset %}
-
-          {{ govukTextarea({
-            name: "title",
-            id: "title",
-            value: form.title,
-            label: {
-              text: titleLabelHtml,
-              classes: "govuk-label--l",
-              isPageHeading: true
+        <input type="hidden" name="_csrf" value="{{ csrfToken }}" />
+        <input type="hidden" name="prisonNumber" value="{{ form.prisonNumber }}" />
+
+        {% set titleLabelHtml %}
+          Describe the goal you want to create for {{ prisonerSummary.firstName }} {{ prisonerSummary.lastName }}
+        {% endset %}
+
+        {% set hintTextHtml %}
+          <p class="govuk-hint">Goals should be SMART, meaning 'specific', 'measurable', 'achievable', 'relevant' and 'time-bound'. SMART goals allow progress to be better measured and evaluated.</p>
+          <p class="govuk-hint">You can add the steps to achieve the goal on the next page.</p>
+        {% endset %}
+
+        {{ govukTextarea({
+          name: "title",
+          id: "title",
+          value: form.title,
+          label: {
+            text: titleLabelHtml,
+            classes: "govuk-label--l",
+            isPageHeading: true
+          },
+          hint: {
+            html: hintTextHtml
+          },
+          errorMessage: errors | findError('title')
+        }) }}
+
+        {% set targetCompletionDateItems = [] %}
+
+        {% for futureGoalTargetDate in futureGoalTargetDates %}
+          {% set targetCompletionDateItems = (targetCompletionDateItems.push({
+            "value": futureGoalTargetDate.value,
+            "text": futureGoalTargetDate.text
+            }), targetCompletionDateItems)
+          %}
+        {% endfor %}
+
+        {% set anotherDateHtml %}
+          {{ govukDateInput({
+            id: "another-date",
+            namePrefix: "another-date",
+            fieldset: {
+              legend: {
+                text: "What date are they aiming to achieve this by?",
+                classes: "govuk-visually-hidden"
+              }
             },
             hint: {
-              html: hintTextHtml
-            },
-            errorMessage: errors | findError('title')
+              text: "For example, 27 3 2007"
+            }
           }) }}
+        {% endset -%}
```

Comment:
Whats the `-%}` syntax here? Is the `-` a typo? or is it some magic secret sauce I've not see before? 😁

Reply:
It just removes whitespace in HTML. Because we're just setting something in the template file, we don't want the whitespace to appear in the rendered HTML DOM. Nothing fancy!